### PR TITLE
Refactor `CLI::PreviewCards#remove` command

### PR DESCRIPTION
**Pull request by mjankowski**
Cherry-picked the reset connection pool stub from another PR, will need rebase assuming that one is merged.

### CODE REVIEW

File: lib/mastodon/cli/preview_cards.rb
Comment (ClearlyClaire) [inline lib/mastodon/cli/preview_cards.rb:1]: Same as the other PR, I'm not too sure about the refactor, and I don't think it should be bundled with the new tests.

Comment (mjankowski) [reply lib/mastodon/cli/preview_cards.rb:1]: Specs only: https://github.com/mastodon/mastodon/pull/27441

**Comment by mjankowski** [comment]
This is one of my oldest PRs where I think there's still good potential for cleanup but where we're at a review impasse. I'm going to attempt to pull out what I think are the remaining useful pieces into smaller PRs, rebasing this one if/when merged, and see how much of it is useful, and then close this one when done.

**Comment by mjankowski** [comment]
Closing as useful portions have been extracted. May re-attempt some of this in other ways in future.

**Pull request closed without merging**